packages/preview-middleware/src/base/telemetry-reporter.ts

```diff
@@ -0,0 +1,41 @@
+import { ClientFactory, SampleRate, initTelemetrySettings } from '@sap-ux/telemetry';
+import modulePackageJson from '../../package.json';
+import { Logger } from '@sap-ux/logger';
+
+const key = 'ApplicationInsightsInstrumentationKeyPLACEH0LDER';
+
+/**
+ * Class handling telemetry initialization and reporting.
+ */
+export class TelemetryReporter {
+    private initialized: boolean;
+    constructor(private eventName: string, private readonly logger: Logger, private layer?: string) {}
+    public async initializeTelemetry() {
+        try {
+            await initTelemetrySettings({
+                consumerModule: modulePackageJson,
+                internalFeature: false,
+                watchTelemetrySettingStore: true,
+                resourceId: key
+            });
+            this.initialized = true;
+        } catch (e) {
+            this.initialized = false;
+            this.logger.error(`Could not initialize telemetry. ${e}`);
+        }
+    }
+
+    public reportTelemetry(data: any): void {
```

Comment:
Would it be good to make this method generic to accept some type depending on the use case? Because essentially data can be anything.

```suggestion
    public reportTelemetry<T extends {}>(data: T): void {
```

Reply:
currently we have less parameters to report , shall we use below ?
```
data: {
        category: string;
        changeType: string;
        controlType: string;
        propertyName: string;
        sapui5Version: string;
    }
```

Reply:
But then when you use more parameters or want to send data that is not on this data object, what do you do? I think we need to have more flexibility here. Or these properties will always stay the same and you can pass only those?

Reply:
ok  lets keep it flexible. I will change it now